OpenSim/Common/Component.h

```diff
@@ -335,56 +367,53 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
      */
     template <typename T>
     friend class ComponentListIterator;
+
+
+    /** Get the complete pathname for this Component to its ancestral Component,
+     *  which is the root of the tree to which this Component belongs.
+     * For example: a Coordinate Components would have a full path name like:
+     *  `/arm26/elbow_r/flexion`. Accessing a Component by its fullPathName from
+     * root is guaranteed to be unique. */
+    std::string getFullPathName() const;
+
+
+    /** Get the relative pathname of this Component with respect to another one */
```

Comment:
Ah so the path name stuff doesn't account for the actual tree; it just deals with strings.